### DISCUSSION TITLE
Cypress keeps crashing

### DISCUSSION THREAD

**Discussion by taydin-git** [cypress-io/cypress#24528] [Questions and Help]
Whenever I run my test cases after a while cypress gets closed itself and gives heap out of memory.
Cypress Version :10.11.0
Node v:16.13.1
[Cypress Crash Report.txt](https://github.com/cypress-io/cypress/files/9932143/Cypress.Crash.Report.txt)

**Comment by lmiller1990** [comment]
Definitely unexpected... your machine is a beast:

> Model: MacBookPro16,1, BootROM 1731.140.2.0.0 (iBridge: 19.16.16067.0.0,0), 8 processors, 8-Core Intel Core i9, 2.4 GHz, 32 GB, SMC 

How many specs/tests? Does it run for a while *then* crash? Can you share a bit more info?

**Reply by lmiller1990** [reply]
Related?

https://github.com/cypress-io/cypress/issues/24509
https://github.com/cypress-io/cypress/issues/24356

**Reply by taydin-git** [reply]
yes same issue

**Comment by taydin-git** [comment]
I am running only one spec and 2 test cases. It happens when I run so many times but this shouldn't happen because I need to run my tests to see failures and successes.

**Reply by lmiller1990** [reply]
This certainly shouldn't be happening. I'm able to run a large suite on my 16GB 2015 macbook just fine, though.

How did you get that report? Normally I do `DEBUG=cypress:* yarn cypress open ...`. You should see lots of logs starting with `cypress:`.

**Reply by taydin-git** [reply]
Actually, when cypress gets shut down this report pop ups by itself.

**Comment by taydin-git** [comment]
[CypressDebugReport.txt](https://github.com/cypress-io/cypress/files/9938584/CypressDebugReport.txt)
  this is the debug mode report

**Reply by lmiller1990** [reply]
Just to check, does the same thing happen with another browser? Trying to narrow this down, it might be a Cypress + Chrome issue (helps with debugging a lot).

Try Chromium, Firefox, Edge/Safari? The reason I ask is I see

```
  cypress:server:util:process_profiler \uc0\u9474     0    \u9474      'Chrome'      \u9474       6       \u9474  '52890, 52922, 52919, 52917, 52939, 52920' \u9474     62.5    \u9474      44.58      \u9474   305.8   \u9474     300.07    \u9474    307.36    \u9474 \
```

In the debug report, which looks like Chrome is consuming a lot the CPU (and possibly memory).

**Reply by taydin-git** [reply]
I will try other browser and will update you but I had the same issue with Electron, as well

**Reply by lmiller1990** [reply]
Stupid question but to clarify, did you try some other trivial projects? Eg - is this problem unique to the website you are testing, or a general "Cypress keeps crashing on my machine, no matter what"?

**Reply by taydin-git** [reply]
hi, I tried with electron and firefox and crashed again. I don't have any trivial project to try. Do you have any suggestion for me?

**Reply by lmiller1990** [reply]
You could try the Cypress kitchensink example:

https://github.com/cypress-io/cypress-example-kitchensink

**Comment by stavares843** [comment]
do you have `numTestsKeptInMemory: 0` in your config file?

**Reply by taydin-git** [reply]
Yes, I do have it in the config file.
